### DISCUSSION TITLE
Training data for spancat

### DISCUSSION THREAD

**Discussion by RandomArnab** [explosion/spaCy#11351] [Help: Other Questions] [answered]
Wanted to try out spancategorizer for a text dataset with overlapping entities but can't seem to find anywhere as to how the training data format looks like. Can someone possibly share a training.jsonl file so as to ease this process?

**Comment by ljvmiranda921** [comment, marked as answer]
Hi @RandomArnab ,

In spaCy v3, we use the [serialized .spacy files](https://spacy.io/api/docbin) instead of JSONL. To prepare the training data for span categorization, you need to assign entities in the `doc.span` attribute. Here's an example below for an example sentence "Welcome to the Bank of China":

 ```python
import spacy
from spacy import displacy
from spacy.tokens import Span

text = "Welcome to the Bank of China."

nlp = spacy.blank("en")
doc = nlp(text)

doc.spans["sc"] = [
    Span(doc, 3, 6, "ORG"), 
    Span(doc, 5, 6, "GPE"),
]
```

To serialize them into a `.spacy` file, you need to collect them inside a [`DocBin` object](https://spacy.io/api/docbin) and call the `to_disk()` method. Something like this:

```
from spacy.tokens import DocBin

doc_bin = DocBin(docs=my_list_of_docs)
doc_bin.to_disk("/some/path/to/train.spacy")
```

You can check some of our spaCy projects to better understand this process. We have one for [overlapping spans](https://github.com/explosion/projects/tree/v3/experimental/ner_spancat_compare).

**Reply by RandomArnab** [reply]
Thanks a lot!

**Reply by geetarajagopal** [reply]
Hello,

I am working on a NLP project using spacy where i am trying to use the SpanCat component. I already used ner but i am unable to get the desired performance. I prepared my training data for SpanCat using the method mentioned in this post and the several other posts including using SpanGroup, however when i train my model using these train.spacy and dev.spacy files, my model does not seem to provide any scores. PFB output while training my model: 
![image](https://user-images.githubusercontent.com/89715112/199318496-e74ee8d9-f38b-4af7-940d-377fb5d1c66e.png)

Below is my code:
![image](https://user-images.githubusercontent.com/89715112/199318785-7ef95b52-e0cf-4be9-9a41-8af2aed58583.png)

I tried different combinations by playing around with the default config file generated for spancat but none of them work. I tried using the spacy files from the projects provided by spacy([experimental](https://github.com/explosion/projects/tree/v3/experimental)/[ner_spancat](https://github.com/explosion/projects/tree/v3/experimental/ner_spancat)) and spancat seems to train properly for those spacy files. 

Can someone please provide me pointers where i might be going wrong in preparing the training data? TIA

**Reply by polm** [reply]
Hey, please do not post the same question in multiple threads. We will follow up in #11731.
